source/adios2/ADIOSTypes.h

```diff
@@ -11,36 +11,169 @@
 #ifndef ADIOS2_ADIOSTYPES_H_
 #define ADIOS2_ADIOSTYPES_H_
 
+/// \cond EXCLUDE_FROM_DOXYGEN
```

Comment:
<c*> is still C++ only . I'd separate it if was the actual C header stddef.h and stdint.h